Libraries/LibWebSocket/WebSocket.cpp

```diff
@@ -42,9 +42,14 @@ void WebSocket::start()
     m_impl->on_connected = [this] {
         if (m_state != WebSocket::InternalState::EstablishingProtocolConnection)
             return;
-        set_state(WebSocket::InternalState::SendingClientHandshake);
-        send_client_handshake();
-        drain_read();
+        if (m_impl->handshake_complete_when_connected()) {
+            set_state(WebSocket::InternalState::Open);
+            notify_open();
+        } else {
+            set_state(WebSocket::InternalState::SendingClientHandshake);
+            send_client_handshake();
+            drain_read();
+        }
```

Comment:
This whole hack is basically to avoid trying to send the HTTP upgrade request for the libcurl backend. And also to avoid trying to call is_eof() and read() on the libcurl backend before we've ever read from it (we only read from the libcurl backend *after* sending the 'connected' notification back to the client).